pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -485,6 +541,45 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Set routers for a particular domain.
+		#[pallet::weight(T::WeightInfo::set_domain_multi_router())]
+		#[pallet::call_index(9)]
+		pub fn set_domain_multi_router(
+			origin: OriginFor<T>,
+			domain: Domain,
+			routers: BoundedVec<T::Router, T::MaxRouterCount>,
```

Comment:
If we also need the `MaxRouterCount`, we not an array instead?

Reply:
I'm not sure I understand your question.

Reply:
The user who call `set_domain_multi_router` ideally needs to put always 3 routers in the call?

Reply:
If so, then `[T::Router, T::MaxRouterCount>]` already force that

Reply:
```
error: generic parameters may not be used in const operations
   --> pallets/liquidity-pools-gateway/src/lib.rs:555:25
    |
555 |             routers: [T::Router; T::MaxRouterCount],
    |                                  ^^^^^^^^^^^^^^^^^ cannot perform const operation using `T`
 ```
 
 I propose to go back to using the `BoundedVec` and ensure that it's length is always `T::MaxRouterCount`.

Reply:
Ouch, I forgot about that is not constant. Ideally you should be able to write `T::MaxRouterCount::get()`, but any method from a trait can not be `const`, that's a pity.

Nevertheless, there is still a solution, which is adding a `const MaxRouterCount: u32;` to the Config trait. But again, what you think is better!